x/ccv/provider/keeper/msg_server.go

```diff
@@ -309,16 +293,55 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// RegisterConsumer registers a consumer chain
+func (k msgServer) RegisterConsumer(goCtx context.Context, msg *types.MsgRegisterConsumer) (*types.MsgRegisterConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumerId := strconv.FormatUint(k.Keeper.FetchAndIncrementConsumerId(ctx), 10)
+
+	k.Keeper.SetConsumerIdToRegistrationRecord(ctx, consumerId, *msg.RegistrationRecord)
+	k.Keeper.SetConsumerIdToOwnerAddress(ctx, consumerId, msg.Signer)
+	k.Keeper.SetConsumerIdToPhase(ctx, consumerId, Registered)
+
+	return &types.MsgRegisterConsumerResponse{ConsumerId: consumerId}, nil
+}
+
+// InitializeConsumer initializes a consumer chain
+func (k msgServer) InitializeConsumer(goCtx context.Context, msg *types.MsgInitializeConsumer) (*types.MsgInitializeConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumerId := msg.ConsumerId
+
+	phase, found := k.Keeper.GetConsumerIdToPhase(ctx, consumerId)
+	if !found || (phase != Registered && phase != Initialized) {
+		return nil, errorsmod.Wrapf(types.ErrInvalidPhase,
+			"chain with consumer id: %s has to be in its registered or initialized phase", consumerId)
 	}
 
+	ownerAddress, _ := k.Keeper.GetConsumerIdToOwnerAddress(ctx, consumerId)
+	if k.GetAuthority() == msg.Authority {
+		// message is executed as part of governance proposal and hence we change the owner address
```

Comment:
This won't be the provider module authority. `GetAuthority` has the following godoc comment:
```
GetAuthority returns the x/ccv/provider module's authority.
```
Because of this, not sure if a clarification is needed here. Let me know if I misunderstood or feel free to resolve if it's okay.

Reply:
I don't get it. You're saying `GetAuthority` won't be the provider module authority, but the godoc comment states that it is?

Reply:
Good catch. I misread this, sorry. Actually, look [here](https://github.com/cosmos/interchain-security/blob/main/x/ccv/provider/keeper/keeper.go#L38):
```
// address capable of executing gov messages (gov module account)
authority string
```
Maybe I'm missing something here. I'll need to look more into this (something that would come up during E2E testing as well)>